aspnet/mvc/views/forms/sample/final/Controllers/DemoRegisterController.cs

```diff
@@ -0,0 +1,71 @@
+﻿using FormsTH.ViewModels;
+using Microsoft.AspNet.Mvc;
+
+namespace FormsTH.Controllers
+{
+    // This controller is used only to demonstrate working with forms.
+    public class DemoRegController : Controller
+    {
+        static string _message;
+        public IActionResult Register()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Register(RegisterViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                _message = "HttpPost Register success";
```

Comment:
> Removed all the `HttpPost` from this controller

Why? My complaint was about the `static` field.

> to verify selections are working as expected

I'm doubtful but will look at these in the context of the whole document. In general actions for forms are completely stateless (ignoring the user information, if any).